src/base.controller.server.js

```diff
@@ -13,43 +13,58 @@ class BaseController extends ApiController {
     super(model);
     this.filters = ['type', 'deleted'];
   }
+
   index(req, res, next) {
-    let query = {};
     req.query.offset = this.parsePagination(req.query.offset, 0);
     req.query.limit = this.parsePagination(req.query.limit, 100);
     if (req.query.sort) req.query.sort = this.parseSort(req.query.sort);
     if (req.query.filter) req.query = this.parseFilter(req.query);
 
-    /* istanbul ignore else */
-    if (typeof this.model.parseQuery === 'function') {
-      query = this.model.parseQuery(req.query);
-    }
-
-    if (!query.populate || !Array.isArray(query.populate)) query.populate = [];
+    if (typeof this.model.aggregationPipeline === 'function') {
+      return this.model.aggregate(this.model.aggregationPipeline(req.query)).exec(
```

Comment:
I would create two controller functions, controller.`find` and controller.`aggregate` that are called from `find`